### PR TITLE
New home page changes

### PR DESCRIPTION
**Pull request by knowuh**
I closed other pending PRs to open this one unified PR  these PT stories:

* https://www.pivotaltracker.com/story/show/114577553   custom home page content
* https://www.pivotaltracker.com/story/show/114576939   custom search path
* https://www.pivotaltracker.com/story/show/114578353   custom root for teachers
* https://www.pivotaltracker.com/story/show/114578639   home link renamed to 'my classes' (sometimes)

There are some failing tests, and I have to demo this to @scytacki but otherwise this is probably ready for a review.

### CODE REVIEW

**Comment by knowuh** [comment]
**How to test the new Home Page features in HAS and Learn:**

HAS and Learn have two different behaviors for the home page.

HAS wants to show a custom page to anonymous users, but they want that custom page to have the sign-up box, and the regular project header.  In HAS a teacher is redirected to 'getting_started' when they first log in, or when they click on the banner, or visit the root path. In HAS the 'search' button takes the user the full material search page.

In Learn, the custom home page will not show the signup box, or the project header.  It will override all of the portal chrome except for the project footer.  In Learn, the teacher will not be redirected to the 'getting_started' page when they click on the banner (though they will when they first log in). In Learn, the 'search' button will take the user to the custom home page content.  But there will be an 'advanced search' link (linking to '/search/' path) on the custom home page content.

An administrator must configure each portal appropriately.  Here are the settings for each:

**HAS:**
- Add the HAS homepage content to the "Home Page Content" Text area.
- The "Wrap Home Page Content" checkbox should be checked (default)
- "Custom Teacher Home Path" should redirect to "/getting_started" (default)
- The "Custom Search Path" should be blank (default, which will redirect to '/search')

**Learn:**
- Add Ethan's new homepage content to the "Home Page Content" Text area.
- The "Wrap Home Page Content" checkbox should NOT be checked
- Techers homepage should NOT redirect (undo the default "/getting_started")
- Search should link to "/".

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -114,11 +144,11 @@ def recent_activity
...
-    # If there are no active classes assigned then return to the home page
-    if (!current_visitor.has_active_classes?)
-      redirect_to root_path
-      return
-    end
+    # # If there are no active classes assigned then return to the home page
+    # if (!current_visitor.has_active_classes?)
+    #   redirect_to root_path
+    #   return
+    # end
```
Comment (scytacki) [inline app/controllers/home_controller.rb]: Can we just delete this code?

**Comment by scytacki** [comment]
This looks good. I'm going to wait for the tests to pass and and then I'll merge it. However we shouldn't deploy it to staging because @eireland is testing learn.staging with the 1.5.0 release. 
 
One thing missing is the ability for a admin to enter in custom getting_started content, we'll have to do that later. Here is a story for it: https://www.pivotaltracker.com/story/show/115074629

**Comment by scytacki** [comment]
These look like the normal random failures.

**Pull request merged**